multiversx_sdk_wallet/user_wallet.py

```diff
@@ -22,15 +82,67 @@ def decrypt_secret_key_from_file(cls, path: Path, password: str) -> UserSecretKe
         return cls.decrypt_secret_key(key_file_object, password)
 
     @classmethod
-    def decrypt_secret_key(cls, keyfile_object: Any, password: str) -> UserSecretKey:
-        _, buffer = keyfile.load_from_key_file_object(keyfile_object, password)
-        return UserSecretKey(buffer)
+    def decrypt_mnemonic_from_file(cls, path: Path, password: str) -> Mnemonic:
+        with open(path) as f:
+            key_file_object = json.load(f)
+
+        return cls.decrypt_mnemonic(key_file_object, password)
 
-    def save(self, path: Path, address_hrp: str):
-        obj = self.to_keyfile_object(address_hrp)
+    def save(self, path: Path, address_hrp: Optional[str] = None):
+        obj = self.to_dict(address_hrp)
 
         with open(path, 'w') as json_file:
             json.dump(obj, json_file, indent=4)
 
-    def to_keyfile_object(self, address_hrp: str) -> Dict[str, Any]:
-        return keyfile.convert_to_keyfile_object(self.secret_key.buffer, self.public_key.buffer, self.password, self.randomness, address_hrp)
+    def to_dict(self, address_hrp: Optional[str] = None) -> Dict[str, Any]:
+        if self.kind == UserWalletKind.SECRET_KEY.value:
+            if not address_hrp:
+                raise Exception("address_hrp must be provided when kind == 'secretKey'")
+            return self._to_dict_when_kind_is_secret_key(address_hrp)
+
+        return self._to_dict_when_kind_is_mnemonic()
+
+    def _to_dict_when_kind_is_secret_key(self, address_hrp: str) -> Dict[str, Any]:
+        if not self.public_key_when_kind_is_secret_key:
```

Comment:
Conditions using `is None` are more explicit and readable.

```suggestion
        if self.public_key_when_kind_is_secret_key is None:
```